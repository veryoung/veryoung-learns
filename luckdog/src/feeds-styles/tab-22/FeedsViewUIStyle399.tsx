/**
 * @Author: veryoungwan
 * 推荐卡片
 */

import React from 'react';
import { View, Text, StyleSheet } from '@tencent/hippy-react-qb';
import { FeedsTheme, vectorToArray, getTitleRight } from './components/utils';
import FeedsSpliter from '../../components/FeedsSpliter';
import FeedsViewItem from '../FeedsViewItem';
import { Title } from './components';
import FeedsProtect from '../../mixins/FeedsProtect';
import { CommonCardStyle, FeedsUIStyle, FeedsLineHeight, CLICK_STEP, SmallBookCoverLeftTagStyle } from '../../framework/FeedsConst';
import { reportUDS, BeaconReportProps, strictExposeReporter, BusiKey } from '@/luckdog';
import FeedsUtils from '../../framework/FeedsUtils';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { scale } from '../../components/animationStyle';
import { CardLinkType } from '../../entity/card';
import { CommonProps } from '../../entity/common';
import { throttle } from '@/luckbox';
import { TabId } from '../../entity';
import { initReportInfo, getChangeReportInfo, switchNovelPrefetch } from '../../utils/preFetchCard';
import { BookCoverRadiusStyle } from '../../types/card';
import { BookCover } from '../../components/book-font-cover';


// 默认一屏展示的书籍数
const DEFAULT_PER_NUMBER = 8;

const { width } = FeedsUtils.getScreen();
const IMAGE_WIDTH = (width - (12 * 2) - (16 * 2) - (26 * 3)) / 4;
const IMAGE_HEIGHT = (IMAGE_WIDTH * 81) / 60;

interface Props extends BeaconReportProps, CommonProps {
  index: number;
}

interface State {
  iposition: number;
  books: any[];
}

@FeedsProtect.protect
export default class FeedsViewUIStyle399 extends FeedsViewItem<Props> {
  public static getRowType() {
    return 399;
  }

  public scaleAnim = scale(200, 1.015); // 点击动画效果
  public state: State;

  /**
   * title点击的响应
   */
  public onClickTitle = throttle(() => {
    const { clickBlankJump = false } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    if (clickBlankJump) {
      reportUDS(BusiKey.CLICK__BLANK_TO_MORE, this.props);
      this.scaleAnim.start();
    }
  }, CLICK_STEP);

  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'FeedsViewUIStyle399');
    this.state = {
      iposition: 0,
      books: [],
    };

    const { globalConf: { curTabId = TabId.BOTTOM_RECOMM2 } = {}, index = 0 } = props || {};

    // 局部刷新换一换替换上报参数
    strictExposeReporter.addReportDataHandler(curTabId, index, (moreData) => {
      const { linkType: refreshType } = FeedsUtils.getSafeProps(props, 'itemBean.parsedObject', {});
      const { iposition } = this.state;

      // 如果是局部刷新换一换，需要在上报之前获取到动态的report_info
      const reportInfo = refreshType === CardLinkType.CHANGE ? getChangeReportInfo(props, iposition) : {};

      return {
        ...moreData,
        ...reportInfo,
      };
    });
  }

  public UNSAFE_componentWillReceiveProps(nextProps) {
    this.updateShownBooks(nextProps);

    // 上报查看更多
    const { linkType = 0 } = FeedsUtils.getSafeProps(nextProps, 'itemBean.parsedObject', {});
    if (linkType === CardLinkType.MORE) reportUDS(BusiKey.EXPOSE__TO_MORE, this.props);
  }

  public componentDidMount() {
    // 点击标题动画结束后执行的内容
    this.scaleAnim.onHippyAnimationEnd(() => {
      // 执行跳转
      this.clickBlankJumpMore();
    });

    this.updateShownBooks(this.props);
  }

  public componentWillUnmount() {
    this.scaleAnim?.destory();
  }

  /**
   * 更新展示的书籍
   * @param props props或nextProps
   */
  public updateShownBooks = (props) => {
    const { vRes = {} } = FeedsUtils.getSafeProps(props, 'itemBean.parsedObject', {});
    const books = vectorToArray(vRes);
    this.setState({
      books,
      iposition: 0,
    }, () => {
      const { linkType: refreshType } = FeedsUtils.getSafeProps(props, 'itemBean.parsedObject', {});
      // 如果是局部刷新换一换，需要动态设置report_info
      refreshType === CardLinkType.CHANGE && this.setReportInfo(props);
    });
  };

  /**
   * 设置上报参数
   * @param props 卡片的props或nextProps
   */
  public setReportInfo = (props) => {
    const { books } = this.state;
    const { iPerPageNum = DEFAULT_PER_NUMBER } = FeedsUtils.getSafeProps(props, 'itemBean.parsedObject', {});
    // 获得所有的缓存书籍的屏数
    const pageNum = books.length / iPerPageNum;

    initReportInfo(props, pageNum);
  };

  /**
   * 点击查看全部上报
   */
  public doBeaconByClickAll = () => {
    const { linkType = 0 } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    if (linkType === CardLinkType.MORE) {
      reportUDS(BusiKey.CLICK__TO_MORE, this.props);
      return;
    }
  };

  /**
   * 点击换一换上报
   * @param books 当前页的书籍
   * @param iposition 当前页的下标
   */
  public doBeaconBySwitch = () => {
    const { linkType: refreshType, refreshKey = '' } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    const isPreFetch = refreshType === CardLinkType.CHANGE && refreshKey !== '';

    // 点击上报
    reportUDS(BusiKey.CLICK__CARD_CHANGE, this.props, {
      bigdata_contentid: '',
      ext_data1: isPreFetch ? '1' : '0',
    });
  };

  /**
   * 处理跳转动画结束后加载去往更多页面
   * 跳转地址不为空
   * 支持跳转
   */
  public clickBlankJumpMore = () => {
    const { moreLink = '' } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    moreLink && this.loadUrl(moreLink);
  };

  /**
   * 切换猜你想看 换一换
   */
  public switchNovel = () => {
    const { iPerPageNum: num = DEFAULT_PER_NUMBER,
      refreshKey = '' } = FeedsUtils.getSafeProps(this.props, 'itemBean.parsedObject', {});
    let { iposition, books } = this.state;
    const pageIndex = Math.floor(books.length / num) - 1;

    iposition = iposition >= pageIndex ? 0 : iposition += 1;

    switchNovelPrefetch(this.props, iposition, pageIndex, num, (res) => {
      const preFetchBooks = FeedsUtils.getSafeProps(res[refreshKey] || {}, 'parsedObject.vRes', []);
      books = [...books, ...preFetchBooks];
      this.setState({
        books,
      });
    });

    this.setState({
      iposition,
    });

    this.doBeaconBySwitch();
  };

  /**
   * 点击书籍
   * @param {*} book 书籍
   * @param {*} index 书籍下标
   */
  public handleClick = (book) => {
    const { sRefer = '', sResourceId = '' } = book || {};
    sRefer && this.loadUrl(sRefer);

    reportUDS(BusiKey.CLICK__CARD_BOOK, this.props, { book_id: sResourceId });
  };

  public onBookItemLayout = (event, tabIndex, bookIndex, bookId) => {
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.index,
      tabIndex,
      bookIndex,
      bookId,
      rect: event.layout,
    });
  };

  public onTitleLayout = (event) => {
    strictExposeReporter.updateTitleHeight(this.props.index, event.layout.height);
  };

  public renderBooks = (books, change, num) => {
    const view: any[] = [];
    const { itemBean, index } = this.props || {};
    let data;
    const { iposition } = this.state;
    if (change) {
      // TODO: 为什么要序列化
      data = JSON.parse(JSON.stringify(books));
      data = data.slice(iposition * num, (iposition + 1) * num);
    } else {
      data = books;
    }

    strictExposeReporter.updateBookIds(index, iposition, data.map(book => book.sResourceId));
    for (let i = 0; i < data.length; i++) {
      const book = data[i];
      const { sScore = 0, sResourceId = '', sResourceName, sPicUrl, stStatus } = book;
      const leftTag = FeedsUtils.getLeftTagStyle(stStatus, SmallBookCoverLeftTagStyle);
      view.push(<View
        style={styles.item}
        key={sResourceId}
        onLayout={event => this.onBookItemLayout(event, iposition, i, sResourceId)}
        onClick={strictExposeReporter.triggerExpoCheck(() => this.handleClick(book))}
      >
        <View>
          <BookCover
            height={IMAGE_HEIGHT}
            width={IMAGE_WIDTH}
            url={sPicUrl}
            bookID={sResourceId}
            radius={BookCoverRadiusStyle.SMALL}
            sourceFrom={itemBean?.item_id}
            leftTag={leftTag}
          />
          <Text style={styles.bookName} numberOfLines={2}>
            {sResourceName}
          </Text>
        </View>
        {
          sScore ? <View style={styles.textScoreBlock}>
            <Text style={styles.textScore}>{sScore}分</Text>
          </View> : null
        }
      </View>);
    }

    return view;
  };

  public render(): any {
    countReRender(this, 'FeedsViewUIStyle399');
    const { itemBean, globalConf = {} } = this.props;
    const { parsedObject = {}, title } = itemBean || {};
    const { iPerPageNum = DEFAULT_PER_NUMBER, bIsNeedChange = false, linkType = 0 } = parsedObject;
    const { books = [] } = this.state;
    if (books.length === 0) return null;

    const titleRight = getTitleRight(this.props);
    return (
      <View style={[{ ...CommonCardStyle }, { transform: [{ scale: this.scaleAnim }] }]}>
        <FeedsSpliter style={globalConf.style} lineStyle={itemBean?.bottomLineStyle} />
        <View
          collapsable={false}
          onClick={this.onClickTitle}
        >
          <Title
            title={title}
            onLayout={this.onTitleLayout}
            switchNovel={this.switchNovel}
            changeable={bIsNeedChange || linkType !== CardLinkType.MORE} // loadmore的时候不能展示换一换
            parent={this}
            right={titleRight}
            showDot={
              books.slice(0, 4).every(o => !o.sUpdatedNumber || o.sUpdatedNumber === '0')
              && books.slice(4).some(o => o.sUpdatedNumber && o.sUpdatedNumber !== '0')
            }
            doBeaconByClick={this.doBeaconByClickAll}
          />
        </View>
        <View style={styles.wrap}>
          {
            this.renderBooks(books, bIsNeedChange, iPerPageNum)
          }
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  wrap: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingLeft: 3,
    paddingRight: 3,
  },
  item: {
    marginLeft: 13,
    marginRight: 13,
    width: IMAGE_WIDTH,
    marginBottom: 24,
    justifyContent: 'space-between',
  },
  picText: {
    justifyContent: 'flex-start',
  },
  textScoreBlock: {
    flexDirection: 'row',
  },
  textScore: {
    fontFamily: 'PingFangSC-Semibold',
    fontSize: FeedsUIStyle.T1,
    colors: FeedsTheme.SkinColor.N3,
    fontWeight: 'bold',
    lineHeight: FeedsLineHeight.T1,
  },
  bookName: {
    colors: FeedsTheme.SkinColor.A1,
    fontSize: FeedsUIStyle.T1,
    lineHeight: FeedsLineHeight.T1,
    marginBottom: 2,
    marginTop: 4,
  },
});
