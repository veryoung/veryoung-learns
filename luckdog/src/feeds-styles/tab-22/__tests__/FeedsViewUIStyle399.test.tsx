import React from 'react';
import renderer from 'react-test-renderer';

import FeedsViewUIStyle399 from '../FeedsViewUIStyle399';
import { commonProps } from '@test/mocks/props';
import { itemBean } from './mocks/FeedsViewUIStyle399';

describe('<FeedsViewUIStyle399/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        index: 0,
        itemBean: itemBean as any,
      };
      const tree = renderer.create(<FeedsViewUIStyle399 {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
