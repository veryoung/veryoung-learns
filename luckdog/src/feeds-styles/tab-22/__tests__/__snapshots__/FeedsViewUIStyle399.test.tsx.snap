// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedsViewUIStyle399/> render testing render normal 1`] = `
<div
  nativeName="View"
  style={
    Array [
      Object {
        "backgroundColors": Array [
          "white",
          "#23282d",
          "white",
          "rgba(0, 0, 0, 0.8)",
        ],
        "borderRadius": 12,
        "marginBottom": 0,
        "marginHorizontal": 12,
        "marginTop": 16,
      },
      Object {
        "transform": Array [
          Object {
            "scale": AnimationSet {
              "animationId": undefined,
              "animationList": Array [
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
                Object {
                  "animationId": undefined,
                  "follow": true,
                },
              ],
              "onAnimationEndCallback": [Function],
              "onHippyAnimationCancel": [Function],
              "onHippyAnimationEnd": [Function],
              "onHippyAnimationRepeat": [Function],
              "onHippyAnimationStart": [Function],
              "onRNfqbAnimationCancel": [Function],
              "onRNfqbAnimationEnd": [Function],
              "onRNfqbAnimationRepeat": [Function],
              "onRNfqbAnimationStart": [Function],
            },
          },
        ],
      },
    ]
  }
>
  <div
    nativeName="View"
    style={
      Array [
        Object {
          "height": 0,
        },
      ]
    }
  />
  <div
    nativeName="View"
    onClick={[Function]}
    style={
      Object {
        "collapsable": false,
      }
    }
  >
    <div
      nativeName="View"
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        }
      }
    >
      <p
        nativeName="Text"
        numberOfLines={1}
        style={
          Array [
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 20,
              "maxWidth": 252,
              "paddingBottom": 14,
              "paddingTop": 14,
            },
            Object {},
          ]
        }
        text="都市推荐"
      >
        都市推荐
      </p>
      <div
        className="change-view"
        nativeName="View"
        onClick={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "paddingBottom": 14,
            "paddingTop": 14,
            "position": "relative",
          }
        }
      >
        <p
          nativeName="Text"
          numberOfLines={1}
          style={
            Array [
              Object {
                "colors": Array [
                  "#000000",
                  "#686D74",
                  "#000000",
                  "#ffffff",
                ],
                "fontSize": 12,
                "lineHeight": 16,
              },
              Object {},
            ]
          }
          text="换一换"
        >
          换一换
        </p>
      </div>
    </div>
  </div>
  <div
    nativeName="View"
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "paddingLeft": 3,
        "paddingRight": 3,
      }
    }
  >
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F482%2F22217482%2Ft3_22217482.jpg&bookid=1122217482"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="系统让我去算命"
        >
          系统让我去算命
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="8.5分"
        >
          8.5分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F66%2F15036066%2Ft3_15036066.jpg&bookid=1115036066"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="超能重工"
        >
          超能重工
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="8.0分"
        >
          8.0分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F570%2F29940570%2Ft3_29940570.jpg&bookid=1129940570"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="开局签到就无敌"
        >
          开局签到就无敌
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="7.3分"
        >
          7.3分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F913%2F25304913%2Ft3_25304913.jpg&bookid=1125304913"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="我就这样出名了"
        >
          我就这样出名了
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="8.0分"
        >
          8.0分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F751%2F31673751%2Ft3_31673751.jpg&bookid=1131673751"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="以综艺之名"
        >
          以综艺之名
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="7.0分"
        >
          7.0分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F516%2F25848516%2Ft3_25848516.jpg&bookid=1125848516"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="败家导演"
        >
          败家导演
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="8.6分"
        >
          8.6分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F852%2F36565852%2Ft3_36565852.jpg&bookid=1136565852"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="开局一吨黄金被学姐曝光了"
        >
          开局一吨黄金被学姐曝光了
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="8.1分"
        >
          8.1分
        </p>
      </div>
    </div>
    <div
      nativeName="View"
      onClick={[Function]}
      onLayout={[Function]}
      style={
        Object {
          "justifyContent": "space-between",
          "marginBottom": 24,
          "marginLeft": 13,
          "marginRight": 13,
          "width": 56.5,
        }
      }
    >
      <div
        nativeName="View"
        style={Object {}}
      >
        <div
          nativeName="View"
          style={
            Array [
              Object {
                "position": "relative",
              },
              Object {
                "height": 76.275,
                "width": 56.5,
              },
            ]
          }
        >
          <img
            alt=""
            nativeName="Image"
            reportData={
              Object {
                "sourceFrom": "2_17_1616377703882_2751417393",
              }
            }
            src="https://cdn.bookimg.html5.qq.com/pic?format=2&flag=3&picurl=http%3A%2F%2Fwfqqreader.3g.qq.com%2Fcover%2F29%2F36249029%2Ft3_36249029.jpg&bookid=1136249029"
            style={
              Object {
                "0": Object {
                  "borderRadius": 6,
                  "height": 76.275,
                  "width": 56.5,
                },
                "1": Object {
                  "borderColors": Array [
                    "#e5e5e5",
                    "#31363b",
                    "#00000010",
                    "#ffffff10",
                  ],
                  "borderWidth": 0.5,
                },
              }
            }
          />
        </div>
        <p
          nativeName="Text"
          numberOfLines={2}
          style={
            Object {
              "colors": Array [
                "#242424",
                "#686d74",
                "#242424",
                "white",
              ],
              "fontSize": 12,
              "lineHeight": 16,
              "marginBottom": 2,
              "marginTop": 4,
            }
          }
          text="都市：开局扫码赚了一个亿"
        >
          都市：开局扫码赚了一个亿
        </p>
      </div>
      <div
        nativeName="View"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <p
          nativeName="Text"
          style={
            Object {
              "colors": Array [
                "#FF796E",
                "#794543",
                "#FF796E",
                "#794543",
              ],
              "fontFamily": "PingFangSC-Semibold",
              "fontSize": 12,
              "fontWeight": "bold",
              "lineHeight": 16,
            }
          }
          text="7.2分"
        >
          7.2分
        </p>
      </div>
    </div>
  </div>
</div>
`;
